src/formats/png.zig

```diff
@@ -36,7 +37,20 @@ pub const DefaultProcessors = reader.DefaultProcessors;
 pub const DefaultOptions = reader.DefaultOptions;
 pub const required_temp_bytes = reader.required_temp_bytes;
 
+pub const Header = struct {
+    width: u32,
+    height: u32,
+    bits_per_sample: u5, // goes up to 16, u5 is enough
+    color_type: types.ColorType,
+    compression_method: types.CompressionMethod,
+    filter_method: types.FilterMethod,
+    interlace_method: types.InterlaceMethod,
+};
+
 pub const PNG = struct {
+    header: Header = undefined,
```

Comment:
In the case of PNG, we should not store the `header` and `filter_choice` but just pass them to the write() functions, because the reader is not filling those fields, they would be only used when writing and be confusing for those reading the image and not having the information. 

Sorry for the confusion :)

Reply:
Then I don't understand what you wanted me to do... should the PNG struct store anything? if not what am I meant to do? What kind of signature should a write function have?

Reply:
Alright, i'll change time next time i can code